app/controllers/refuels_controller.rb

```diff
@@ -1,2 +1,81 @@
 class RefuelsController < ApplicationController
+  before_action :params_refuel, only: %i[create]
+  before_action :find_vehicle, only: %i[index create new]
+  before_action :find_fuels, only: %i[new create]
+
+  def index
+    @refuels = Refuel.where(vehicle_id: params[:vehicle_id]).order(created_at: :desc)
```

Comment:
```suggestion
    @refuels = Refuel.where(vehicle_id: params[:vehicle_id]).order(:created_at).limit(10)
```
Because we don't need the full index in the final version, you can directly select the 10 last refuels here, and change the order. Then you can remove the commands **take(10).reverse** (3x) in the HTML file.
This way will be faster, you avoid to load 3 full arrays in the method **last_ten_refuels_info**, you only load the useful set of data.

Reply:
I understand your concerns but I wanted to try to implement a slider input to change how many refuels are displayed in the graph, that could be a way to have fun with js, what do you think?

Reply:
The slider is a good idea.
Just use a step of 5 or 10, and add an event on mouseup only, to limit the reloads.
Do you have an idea to redraw the graph without refreshing the full page ?
For me, the datasets are popuated when the page is loaded. So, if you change the slider value, you have to reload the page to update the datasets.
In this case, each refresh will load the full index and populate the 3 full arays. So it's better to send the slider value to the controller to limit the records returned by the controller.

But maybe I'm wrong and you can redraw the graph without refreshing the full page.

Last thing, I made a mistake with my suggestion (if you need it), it's not **limit(n)** but **last(n)** to retrun the nth last records.